bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/MultiProbeStrategy.scala

```diff
@@ -0,0 +1,128 @@
+      }).map(a => ctxFixed.withPositionAngle(ctx.getPositionAngle.add(a, Angle.Unit.DEGREES)))
+    }
+
+    // For each ObsContext, perform the selection and return it as a future.
+    val noStrategy: Option[AgsStrategy.Selection] = None
+    def ctxSelect(cCtx: ObsContext): Future[(ObsContext, Option[AgsStrategy.Selection])] =
+      Future.fold(strategies.map(_.select(cCtx, mt)))(cCtx -> noStrategy) {
+        case ((_, resOpt), curOpt) =>
+          cCtx -> curOpt.map(cur => AgsStrategy.Selection(cur.posAngle, resOpt.fold(cur.assignments)(cur.assignments ++ _.assignments)))
+      }
+
+    // We need a way to compare AGS selections to determine which is the best across the list of generated
+    // ObsContexts. We assign a 4-tuple to each selection in such a way that the minimum of the 4-tuples represents
+    // the best selection.
+    // 1. Quality index is the quality converted to an int, and then summed over all targets.
+    // 2. Vignetting index is vignetting for each vignetting guide probe, summed across all.
+    // 3. Best quality across all targets. Used because for magnitude, we only want best quality class in comparison.
+    // 4. Magnitude is magnitude of the brightest star in the assignment in the best quality class.
+    def selValue(cCtx: ObsContext, sel: AgsStrategy.Selection): (Int, Double, Int, Double) = {
+      val qualitiesByTarget = sel.assignments.collect {
+        case AgsStrategy.Assignment(gp: ValidatableGuideProbe, t) => t -> analyze(cCtx, mt, gp, t)
+      }.collect {
+        case (t, Some(v)) => t -> AgsGuideQuality.All.indexOf(v.quality)
+      }.toMap
+      val (bestQuality, qualityIdx) = {
+        val qualities = qualitiesByTarget.values
+        (qualities.min, qualities.sum)
+      }
+
+      val vignettingIdx = sel.assignments.collect {
+        case AgsStrategy.Assignment(gp: SingleProbeStrategy.VProbe, t) => gp.calculator(cCtx).calc(t.coordinates)
+      }.sum
+
+      val brightestMag = sel.assignments.collect {
+        case AgsStrategy.Assignment(_, t) if qualitiesByTarget.get(t).exists(_ === bestQuality) =>
+          probeBands.extract(t).map(_.value)
+      }.collect {
+        case Some(v) => v
+      }.minimum.getOrElse(Double.MaxValue)
+
+      (qualityIdx, vignettingIdx, bestQuality, brightestMag)
+    }
+
+    // Turn into a Future[List[(ObsContext, Option[AgsStrategy.Selection])]], collect the results that have
+    // a defined selection while calculating their value, and then return the one (if one exists) of minimum value.
+    Future.sequence(ctxToFixedList.map(ctxSelect))
+      .map(_.collect { case (cCtx, Some(sel)) => (cCtx, sel, selValue(cCtx, sel)) }.minimumBy(_._3).map(_._2))
+  }
+
+
+  // From the left, take the band that is the superset of the most bands.
+  // This should but may not contain all bands, as BandsList is a sealed trait.
+  override lazy val probeBands = strategies.map(_.probeBands).reduceLeft((sup,cur) => {
```

Comment:
I don't think this covers the (highly hypothetical) case where you have RBands + K. If necessary we can expand BandsList